lib/vcs.ml

```diff
@@ -146,6 +146,9 @@ let git_tag ~dry_run r ~force ~sign ~msg ~commit_ish tag =
 let git_delete_tag ~dry_run r tag =
   run_git_quiet ~dry_run r Cmd.(v "tag" % "-d" % tag)
 
+let git_submodule_update ~dry_run r =
+  run_git_quiet ~dry_run r Cmd.(v "submodule" % "update" % "--init")
```

Comment:
Should it also use `--recursive` like done in `ocaml-ci`/`ocurrent`?

See https://github.com/ocurrent/ocurrent/blob/a8909d5669bcb2428ccdc27591011c1e5897dbc3/plugins/git/cmd.ml#L48

Reply:
Ah sorry I should've read the original description of this PR, you already suggested it ^^"

Reply:
As I mentioned in the PR message, it might make sense to do so but I'd be curious to get some feedback before making this the default. I think if we start relying to heavily on submodules that way, we could run into problems such as:
I have two submodules a and b and both of them have a the same c submodule. If we pull recursively c will be duplicated and assuming a, b and c use dune, this might fail.

If we don't recursively pull submodules, the use can chose to include c himself.

I feel like it is easier to work around those issues if `--recursive` isn't the default or is at least configurable. Making it configurable might prove a bit of a headache from a user interface perspective and I'd rather not dive into it before someone actually expresses the need for it!

Do you actually need the `--recursive` in one of your project?

Reply:
> Do you actually need the `--recursive` in one of your project?

No it was just a small remark, no worries. It worked great for me so far, I'll post a proper comment once CI is green on the PR I've made using it.